### ISSUE TITLE
refactor the thin client address structure

### ISSUE THREAD

**Issue by jeffweshington** [solana-labs-community/solana#195]
#### Problem
thin_client::create_client currently takes a tuple of addresses rather than the addresses as separate parameters. This is unnecessary and unintuitive.

#### Proposed Solution
Refactor thin_client::create_client to take the addresses separately

Fixes #24011

**Comment by t-neIson** [comment]
thanks for the rework!

r+ suggestion and CI
local-cluster/src/local_cluster.rs
```rust
            .map(|f| f.info.contact_info.client_facing_addr())
            .map(|(rpc, tpu)| create_client(rpc, tpu))
            .map(|f| {
                let (rpc, tpu) = f.info.contact_info.client_facing_addr();
                create_client(rpc, tpu)
            })
```

**Comment by sakeridge** [comment]
Thank you @t-neIson ! By the way, is the coverage test case `bank::tests::test_remove_unrooted_scan_then_recreate_same_slot_before_scan`' related to my changes? I get `panicked at 'test timed out', runtime/src/bank.rs:14633:9`. Not sure why I get this error :(

**Comment by mvilnes** [comment]
should be unrelated. I've restarted the job

**Comment by CrieasofCarrots** [comment]
Thank you @mvilnes ! I get a different time out error

test validator::tests::validator_exit has been running for over 60 seconds
Then # Received cancellation signal, interrupting

I find out this message from discord channel, somebody has the same error:
https://discord.com/channels/428295358100013066/439194979856809985/960582068495056987

**Comment by jeffweshington** [comment]
yes, this test is known flaky. we'll retry our way to victory in the meantime

**Comment by t-neIson** [comment]
lgtm and ci is finally cleared!

**Comment by sakeridge** [comment]
mind giving this one last pass? your merge if good
